### PR TITLE
Name the ImageNet pseudo-label teacher checkpoints and correct stored dtype

### LINKED ISSUE #25746: Question about the ImageNet pseudo-labeled depth source
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Question about the ImageNet pseudo-labeled depth source: https://docs.ultralytics.com/datasets/depth/imagenet-pseudo

The page says the depth targets were produced by "a pretrained Depth Anything V3 teacher model (monocular and metric)" and saved as float32 `.npy` arrays in meters, but it doesn't say which checkpoint. DA3 ships several: the main series (Small/Base/Large/Giant), DA3Metric-Large, DA3Mono-Large, and DA3Nested-Giant-Large, plus the retrained `-1.1` variants.

Two things I'd like to pin down:

1. Which checkpoint was it? 
2. How were meters obtained? DA3Metric-Large's README gives `metric_depth = focal * net_output / 300.`, where `focal` is in pixels from the camera intrinsics — and ImageNet images have no intrinsics. 

Asking because I want to apply the same recipe to my own unlabeled domain images, and the answer changes what I should store.

### Additional

_No response_

### PR DESCRIPTION
**Pull request by Bovey0809**
The ImageNet pseudo-labeled depth page said the targets came from "a Depth Anything V3 teacher model (monocular and metric)" saved as "float32" arrays, which is not reproducible as written and wrong on the dtype. Raised in #25695 and #25746 by a user wanting to apply the same recipe to their own unlabeled images.

Verified against the generated labels before writing this up: re-running the checkpoints on ImageNet images whose stored `.npy` survives shows an affine fit from the `DA3MONO-LARGE` prediction to the stored label with a 0.001-0.013 m median residual on 0.4-12 m maps (float16 rounding level), and a `focal/300`-scaled metric anchor misses by 0.4-2.4x, ruling out any intrinsics-based scaling. All sampled label files are float16.

- names the two checkpoints and their roles: `DA3MONO-LARGE` for structure, `DA3METRIC-LARGE` for scale, joined by a per-image robust affine fit
- states that no camera intrinsics are involved, which is what the user needed for uncalibrated images
- corrects float32 to float16 for the stored maps, while keeping float32 as the documented format for custom datasets
- notes the scale-error consequence and why scale-invariant SILog training plus median-aligned validation absorbs it
- cites Depth Anything 3 instead of the Depth Anything v1 paper

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Updated the ImageNet pseudo-labeled depth documentation to identify the Depth Anything 3 checkpoints and accurately describe pseudo-label generation and storage.

### 📊 Key Changes
- Names `DA3MONO-LARGE` as the source of detailed scene structure and `DA3METRIC-LARGE` as the source of metric scale.
- Documents the per-image robust affine fit between the two predictions and clarifies that camera intrinsics are not used.
- Corrects stored `.npy` label maps from float32 to float16, while noting that the loader converts them to float32 and custom datasets should continue using float32.
- Explains that scale errors in pseudo-labels are addressed during training and validation through SILog loss, gradient matching, and median alignment.
- Replaces the Depth Anything v1 citation and acknowledgements with a Depth Anything 3 reference.

### 🎯 Purpose & Impact
- Users can reproduce the documented pseudo-labeling workflow with the named checkpoints on uncalibrated images and use the correct stored dtype and citation.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Bovey0809, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This automated message confirms your contribution was received, and an Ultralytics engineer will assist with the review. To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: docs/en/datasets/depth/imagenet-pseudo.md [hunk 1]
```diff
@@ -25,17 +25,20 @@ The pseudo-labeled ImageNet source consists of the ImageNet-1K training images w
 
 ### How the pseudo-labels are generated
 
-Because ImageNet ships without depth annotations, depth targets are produced offline rather than measured:
+Because ImageNet ships without depth annotations, depth targets are produced offline rather than measured. Two Depth Anything 3 checkpoints are combined, one for scene structure and one for absolute scale:
 
-1. Each ImageNet training image is run through a pretrained Depth Anything V3 teacher model to predict a per-pixel metric depth map.
-2. The teacher prediction is saved as a `.npy` float32 array in meters, following the [Ultralytics depth dataset format](index.md), and paired with its source image by file stem.
-3. The pseudo-labeled pairs are then added as one component of a larger training mix, where a student YOLO26-Depth model learns to match the teacher while also training on real ground-truth sources.
+1. Each ImageNet training image is run through `depth-anything/DA3MONO-LARGE`, which predicts a highly detailed depth map that is only defined up to an unknown per-image scale and shift, and through `depth-anything/DA3METRIC-LARGE`, whose output is coarser but in real units.
```
Comment (UltralyticsAssistant) [inline docs/en/datasets/depth/imagenet-pseudo.md:30]: 💡 **MEDIUM**: `DA3METRIC-LARGE` is not itself a meter-valued anchor in the official release: its README documents converting the raw output with `focal * net_output / 300`, while noting that `DA3NESTED-GIANT-LARGE` is already in meters. Describing this standalone checkpoint as producing real-unit depth, then claiming no intrinsics are involved, leaves users unable to reproduce calibrated meter labels. Please document the exact raw-output normalization or assumed focal used here, or describe the result as relative rather than metric.

File: docs/en/datasets/depth/imagenet-pseudo.md [hunk 2]
```diff
@@ -25,17 +25,20 @@ The pseudo-labeled ImageNet source consists of the ImageNet-1K training images w
...
+2. A per-image robust affine fit maps the monocular prediction onto the metric one, `label = a * mono + b`. Every per-pixel detail therefore comes from the monocular checkpoint, while the metric checkpoint only sets the two scalars that place the map on the meter axis. No camera intrinsics are involved, so images without calibration can be labeled.
```
Comment (UltralyticsAssistant) [inline docs/en/datasets/depth/imagenet-pseudo.md:31]: 💡 **MEDIUM**: “Per-image robust affine fit” is still underspecified for the stated reproducibility goal. The estimator and robustness parameters, valid-pixel/sky mask, image preprocessing, and resolution used for fitting can materially change `a` and `b`; a user rerunning the named checkpoints with ordinary least squares or a different mask will not necessarily regenerate these labels. Please specify those details or link the label-generation implementation.

File: docs/en/datasets/depth/imagenet-pseudo.md [hunk 3]
```diff
@@ -25,17 +25,20 @@ The pseudo-labeled ImageNet source consists of the ImageNet-1K training images w
...
+3. The result is saved as a `.npy` array in meters, following the [Ultralytics depth dataset format](index.md), and paired with its source image by file stem. These maps are stored as float16 to halve the on-disk footprint across 1.28M files; the dataset loader casts them to float32 on read, and float32 remains the documented format for custom depth datasets.
+4. The pseudo-labeled pairs are then added as one component of a larger training mix, where a student YOLO26-Depth model learns to match the teacher while also training on real ground-truth sources.
+
+Because the scale comes from a prediction rather than a measurement, each map can carry a global scale error. YOLO26-Depth trains with a scale-invariant log (SILog) loss plus gradient matching and validates with median alignment, so a per-image scale offset in the pseudo labels is largely absorbed.
```
Comment (UltralyticsAssistant) [inline docs/en/datasets/depth/imagenet-pseudo.md:35]: 💡 **MEDIUM**: The preceding fit includes an additive shift `b`, but the documented SILog/gradient losses operate in log-depth and median alignment primarily absorbs multiplicative scale. A nonzero additive depth offset is not generally removed by either mechanism and can change relative errors and log gradients, especially at short ranges. Please distinguish the multiplicative scale error from the affine shift and explain why the shift is negligible or how it is handled.

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Reviewed the complete documentation diff and the related depth loader/loss behavior. The dtype correction and checkpoint naming are consistent with the repository, but the metric semantics, fit specification, and claimed absorption of affine-label errors need clarification before this is reproducible and technically accurate.

<details><summary>💬 Posted 3 inline comments</summary>

- 💡 **MEDIUM** `docs/en/datasets/depth/imagenet-pseudo.md:30` `DA3METRIC-LARGE` is not itself a meter-valued anchor in the official release: its README documents converting the raw output with `focal * net_output / 300`, while noting that `DA3NESTED-GIANT-LARGE` is already in meters. Describing this standalone checkpoint as producing real-unit depth, then claiming no intrinsics are involved, leaves users unable to reproduce calibrated meter labels. Please document the exact raw-output normalization or assumed focal used here, or describe the result as rel…
- 💡 **MEDIUM** `docs/en/datasets/depth/imagenet-pseudo.md:31` “Per-image robust affine fit” is still underspecified for the stated reproducibility goal. The estimator and robustness parameters, valid-pixel/sky mask, image preprocessing, and resolution used for fitting can materially change `a` and `b`; a user rerunning the named checkpoints with ordinary least squares or a different mask will not necessarily regenerate these labels. Please specify those details or link the label-generation implementation.
- 💡 **MEDIUM** `docs/en/datasets/depth/imagenet-pseudo.md:35` The preceding fit includes an additive shift `b`, but the documented SILog/gradient losses operate in log-depth and median alignment primarily absorbs multiplicative scale. A nonzero additive depth offset is not generally removed by either mechanism and can change relative errors and log gradients, especially at short ranges. Please distinguish the multiplicative scale error from the affine shift and explain why the shift is negligible or how it is handled.
</details>

**Comment by Bovey0809** [comment]
@Y-T-G

**Comment by glenn-jocher** [comment]
Closing under this batch's high bug-fix gate. This is a factual rewrite of internal pseudo-label generation, not a reproducible package defect, and the repository has no public artifact or owner-level source from which the new checkpoint, affine-fit, float16-storage, and training-loss claims can be independently verified. Those details should land from the dataset-generation source of truth (or with verifiable provenance), not as an isolated narrative correction here.

**Comment by UltralyticsAssistant** [comment]
🎉 This PR has been merged—thank you @Bovey0809, with contributions from @glenn-jocher!

> “Alone we can do so little; together we can do so much.” — Helen Keller

The updated documentation now clearly identifies the Depth Anything 3 checkpoints, explains pseudo-label generation for uncalibrated images, documents the float16 storage and float32 loading behavior, and replaces the outdated citation and acknowledgements.
